datafusion/core/tests/dataframe/dataframe_functions.rs

```diff
@@ -870,3 +870,22 @@ async fn test_fn_array_to_string() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_fn_string_to_array() -> Result<()> {
+    let expr = string_to_array(lit("abc##def##ghi"), lit("##"), lit("!"));
```

Comment:
Some test cases are already covered in `array.slt`

Reply:
Firstly, thanks for review. Addressed.